### PR TITLE
[13.0][MIG] mgmtsystem_action: Migration to 13.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa TT20481

### CODE REVIEW

File: mgmtsystem_action/models/mgmtsystem_action.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+    @api.depends("date_closed", "create_date")
+    def _compute_number_of_days_to_close(self):
+        for action in self:
+            action.number_of_days_to_close_open = action._elapsed_days(
+                action.create_date, action.date_closed
+            )
+
+    @api.model
+    def _stage_groups(self, stages=None, domain=None, order=None):
```
Comment (Tardo) [inline mgmtsystem_action/models/mgmtsystem_action.py:103]: stages and domain are never used... is this premeditated?

File: mgmtsystem_action/models/mgmtsystem_action.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        return self.env["mgmtsystem.action.stage"].search([], order=order)
+
+    @api.model
```
Comment (Tardo) [inline mgmtsystem_action/models/mgmtsystem_action.py]: Try use 'api.model_create_multi'

File: mgmtsystem_action/models/mgmtsystem_action.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    def create(self, vals):
+        if vals.get("reference", _("New")) == _("New"):
+            Sequence = self.env["ir.sequence"]
+            vals["reference"] = Sequence.next_by_code("mgmtsystem.action")
+        action = super(MgmtsystemAction, self).create(vals)
```
Comment (Tardo) [inline mgmtsystem_action/models/mgmtsystem_action.py]:
```suggestion
        action = super().create(vals)
```

File: mgmtsystem_action/models/mgmtsystem_action.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            return True
+        return False
+
+    @api.model
+    def _get_stage_open(self):
+        return self.env.ref("mgmtsystem_action.stage_open")
+
+    def case_open(self):
+        """Opens case."""
+        # TODO smk: is this used?
```
Comment (Tardo) [inline mgmtsystem_action/models/mgmtsystem_action.py:172]: is this used?

File: mgmtsystem_action/data/email_template.xml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+<p>
+  This is an automated message to remind you that the
+  following action is due to be completed in 10 days (${object.date_deadline}):
+
+  <b><a href="${object.get_action_url()}">${object.name}</a></b>
+</p>
+
+<p>
+  Thank you and have a nice day.<br />
```
Comment (Tardo) [inline mgmtsystem_action/data/email_template.xml]:
```suggestion
  Thank you and have a nice day.<br>
```

File: mgmtsystem_action/data/email_template.xml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  --<br />
```
Comment (Tardo) [inline mgmtsystem_action/data/email_template.xml]:
```suggestion
  --<br>
```

File: mgmtsystem_action/reports/mgmtsystem_action_report.xml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        </field>
+    </record>
+
+    <!-- Management system Action Report Action -->
+
+    <record id="action_report_mgmtsystem_action" model="ir.actions.act_window">
+        <field name="name">Management System Analysis</field>
+        <field name="res_model">mgmtsystem.action.report</field>
+        <field name="view_mode">pivot,graph</field>
+        <field name="context">{"search_default_year":1,"search_default_User":1,"search_default_This Month":1,'group_by_no_leaf':1,'group_by':[]}</field>
```
Comment (Tardo) [inline mgmtsystem_action/reports/mgmtsystem_action_report.xml]: This seems that isn't working...

**Review by Tardo — CHANGES_REQUESTED** [review]

File: mgmtsystem_action/models/mgmtsystem_action.py (hunk 1 above)
Comment (ernestotejeda) [reply mgmtsystem_action/models/mgmtsystem_action.py:103]: That method is the one specified in the group_expand argument of the stage_id field and must to have all those arguments

File: mgmtsystem_action/models/mgmtsystem_action.py (hunk 2 above)
Comment (ernestotejeda) [reply mgmtsystem_action/models/mgmtsystem_action.py]: Why should i do that?

File: mgmtsystem_action/models/mgmtsystem_action.py (hunk 4 above)
Comment (ernestotejeda) [reply mgmtsystem_action/models/mgmtsystem_action.py:172]: Apparently it's used here: https://github.com/OCA/management-system/blob/12.0/mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py#L244

File: mgmtsystem_action/reports/mgmtsystem_action_report.xml (hunk 7 above)
Comment (ernestotejeda) [reply mgmtsystem_action/reports/mgmtsystem_action_report.xml]: I also think so, i'm going to remove it

**Comment by ernestotejeda** [comment]
Changes done

File: mgmtsystem_action/models/mgmtsystem_action.py (hunk 2 above)
Comment (Tardo) [reply mgmtsystem_action/models/mgmtsystem_action.py]: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0

Comment (ernestotejeda) [reply mgmtsystem_action/models/mgmtsystem_action.py]: ok, done

**Comment by ernestotejeda** [comment]
changes done

File: mgmtsystem_action/models/mgmtsystem_action.py [hunk 8]
```diff
@@ -105,14 +103,15 @@ def _compute_number_of_days_to_close(self):
     def _stage_groups(self, stages=None, domain=None, order=None):
         return self.env["mgmtsystem.action.stage"].search([], order=order)
 
-    @api.model
+    @api.model_create_multi
     def create(self, vals):
```
Comment (pedrobaeza) [inline mgmtsystem_action/models/mgmtsystem_action.py]: s/vals/vals_list

File: mgmtsystem_action/models/mgmtsystem_action.py [hunk 9]
```diff
@@ -105,14 +103,15 @@ def _compute_number_of_days_to_close(self):
...
-            vals["reference"] = Sequence.next_by_code("mgmtsystem.action")
-        action = super(MgmtsystemAction, self).create(vals)
-        self.send_mail_for_action(action)
-        return action
+        for one_vals in vals:
+            if one_vals.get("reference", _("New")) == _("New"):
+                Sequence = self.env["ir.sequence"]
+                one_vals["reference"] = Sequence.next_by_code("mgmtsystem.action")
+        actions = super().create(vals)
+        self.send_mail_for_action(actions)
```
Comment (pedrobaeza) [inline mgmtsystem_action/models/mgmtsystem_action.py]: `actions.send_mail_for_action()`

see above

File: mgmtsystem_action/models/mgmtsystem_action.py [hunk 10]
```diff
@@ -130,9 +129,10 @@ def _check_stage_id(self):
                 rec.date_closed = fields.Datetime.now()
 
     @api.model
-    def send_mail_for_action(self, action, force_send=True):
+    def send_mail_for_action(self, actions, force_send=True):
```
Comment (pedrobaeza) [inline mgmtsystem_action/models/mgmtsystem_action.py]: Remove `actions` argument and use `self`.

File: mgmtsystem_action/models/mgmtsystem_action.py [hunk 11]
```diff
@@ -130,9 +129,10 @@ def _check_stage_id(self):
                 rec.date_closed = fields.Datetime.now()
 
     @api.model
```
Comment (pedrobaeza) [inline mgmtsystem_action/models/mgmtsystem_action.py]: Remove this

File: mgmtsystem_action/reports/mgmtsystem_action_report.py [hunk 12]
```diff
@@ -34,7 +34,6 @@ class MgmtsystemtActionReport(models.Model):
     stage_id = fields.Many2one("mgmtsystem.action.stage", "Stage", readonly=True)
     system_id = fields.Many2one("mgmtsystem.system", "System", readonly=True)
 
-    @api.model_cr
     def init(self):
         """Display a pivot view of action."""
         tools.drop_view_if_exists(self._cr, "mgmtsystem_action_report")
```
Comment (pedrobaeza) [inline mgmtsystem_action/reports/mgmtsystem_action_report.py:89]: Can you use the same approach than new views for favoring inheritance?

Example: https://github.com/odoo/odoo/blob/c32126da5ac55a03637d82cf520b8655be8546e6/addons/sale/report/sale_report.py#L147

File: mgmtsystem_action/reports/mgmtsystem_action_report.xml [hunk 13]
```diff
@@ -33,10 +33,7 @@
     <record id="action_report_mgmtsystem_action" model="ir.actions.act_window">
         <field name="name">Management System Analysis</field>
         <field name="res_model">mgmtsystem.action.report</field>
-        <field name="view_type">form</field>
         <field name="view_mode">pivot,graph</field>
-        <field name="context">{"search_default_year":1,"search_default_User":1,"search_default_This Month":1,'group_by_no_leaf':1,'group_by':[]}</field>
```
Comment (pedrobaeza) [inline mgmtsystem_action/reports/mgmtsystem_action_report.xml]: Why do you remove default filters?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: mgmtsystem_action/reports/mgmtsystem_action_report.xml (hunk 13 above)
Comment (ernestotejeda) [reply mgmtsystem_action/reports/mgmtsystem_action_report.xml]: Because it doesn't seem to work

Comment (pedrobaeza) [reply mgmtsystem_action/reports/mgmtsystem_action_report.xml]: Well, they work if they are defined as filters in the search view. But if they have disappeared, OK to remove them.

**Comment by ernestotejeda** [comment]
changes done

**Review by chienandalu — CHANGES_REQUESTED** [review]
@ernestotejeda You've got merge conflicts now. Please rebase

**Comment by ernestotejeda** [comment]
merge conflicts resolved

File: .isort.cfg [hunk 14]
```diff
@@ -9,4 +9,4 @@ line_length=88
 known_odoo=odoo
 known_odoo_addons=odoo.addons
 sections=FUTURE,STDLIB,THIRDPARTY,ODOO,ODOO_ADDONS,FIRSTPARTY,LOCALFOLDER
-known_third_party=openupgradelib,setuptools
+known_third_party = mock,openupgradelib,setuptools
```
Comment (pedrobaeza) [inline .isort.cfg]: Why the extra space in between?

**Review by pedrobaeza — APPROVED** [review]

File: .isort.cfg (hunk 14 above)
Comment (ernestotejeda) [reply .isort.cfg]: I had deleted the entire line of 'known_third_party' and ran pre-commit run -a --show-diff-on-failure and that was how that line was automatically created (with extra spaces), anyway I already deleted the extra spaces :-)

Comment (pedrobaeza) [reply .isort.cfg]: black doesn't format .cfg, so it's weird what you say, but let's move on.

**Comment by pedrobaeza** [comment]
@Tardo @chienandalu please finish your reviews

File: mgmtsystem_action/reports/mgmtsystem_action_report.py [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+    )
+    create_date = fields.Datetime("Create Date", readonly=True, index=True)
+    date_open = fields.Datetime("Opening Date", readonly=True, index=True)
+    date_closed = fields.Datetime("Close Date", readonly=True, index=True)
+    date_deadline = fields.Date("Deadline", readonly=True, index=True)
+    user_id = fields.Many2one("res.users", "User", readonly=True)
+    stage_id = fields.Many2one("mgmtsystem.action.stage", "Stage", readonly=True)
+    system_id = fields.Many2one("mgmtsystem.system", "System", readonly=True)
+
+    def _query(
```
Comment (Tardo) [inline mgmtsystem_action/reports/mgmtsystem_action_report.py:37]: This method is the paradise for sql-injection xD ... Needs a big sign saying "Use under your own responsibility". If other uses it on a public method in a bad way... others could abuse it via rpc.

**Review by Tardo — APPROVED** [review]
LGTM :+1:

File: mgmtsystem_action/reports/mgmtsystem_action_report.py (hunk 15 above)
Comment (pedrobaeza) [reply mgmtsystem_action/reports/mgmtsystem_action_report.py:37]: Well, being a private method, the scope is reduced and no external access can exploit, and if you have ORM access, then you may abuse directly without using this method.

**Pull request merged**